src/librustc/ty/mod.rs

```diff
@@ -1585,13 +1585,12 @@ rustc_index::newtype_index! {
     /// declared, but a type name in a non-zero universe is a placeholder
     /// type -- an idealized representative of "types in general" that we
     /// use for checking generic functions.
+    #[derive(HashStable)]
     pub struct UniverseIndex {
         DEBUG_FORMAT = "U{}",
     }
```

Comment:
This should be:
```rust
pub struct UniverseIndex {
    derive [HashStable]
    DEBUG_FORMAT = "U{}",
}
```
But maybe the macro works fine with the attribute too. Do you get an error if you remove the derive here?

Reply:
The compilation breaks when removing it, yes. I guess it does the job. I can add a commit to rectify this.

Reply:
Just keep it like this. It seems the special derive syntax in the macro is unnecessary.